lib/Wikia/src/Service/User/Attributes/UserAttributes.php

```diff
@@ -47,15 +53,32 @@ public function getAttribute( $userId, $attributeName, $default = null ) {
 	}
 
 	private function loadAttributes( $userId ) {
-		if ( !isset( $this->attributes[$userId] ) ) {
-			$this->attributes[$userId] = [];
+
+		if ( isset( $this->attributes[$userId] ) ) {
+			return $this->attributes[$userId];
+		}
+
+		$attributes = $this->loadFromMemcache( $userId );
```

Comment:
Did you consider using https://github.com/Wikia/app/blob/dev/includes/wikia/WikiaDataAccess.class.php#L52 ?